### DISCUSSION TITLE
geoBoundaries API - Updates for 4.0

### DISCUSSION THREAD

**Discussion by DanRunfola** [wmgeolab/geoBoundaries#1550] [Ideas]
Hi all,

Just sticking a quick comment here for any feedback - for the 4.0 release of geoBoundaries, we plan on a major upgrade to our API, moving from a (fragile) PHP/CSV backend to a much more responsive, pre-cached REST API.  Essentially, we have a small script that pre-generates everything now, so a PHP script isn't doing any heavy lifting on-the-fly.

This is necessitating a few changes to the API - the preview of this can be seen at : 
https://wmgeolab.github.io/gbWeb/api.html

Of note, this is not yet live on the operational geoboundaries site, so if you want to test the new API you'll need to replace www.geoboundaries.org with https://wmgeolab.github.io/gbWeb/

The current plan is to deprecate the old API in May of 2022, to give users plenty of time to migrate.  However, we'll only be supporting new boundaries (4.0 onward) in the new API structure.

None of this is set in stone just yet, but the old API is definitely not up to the task :)

-Dan

**Comment by leeberryman** [comment]
@DanRunfola is https://www.geoboundaries.org/api/gbID/[geoBoundaryID]/ how does the historical request work?  Does this only return a result if the particular boundary has been archived and is no longer "current"?  I only get 404s but I might have only used current IDs

**Comment by DanRunfola** [comment]
These are for the full shapefiles, not boundaries, is my guess at the issue
here.  I.e., a gb-id for the full ADM2/AFG would work, but not an
individual geometry within that.
